addons/website_event_booth/controllers/event_booth.py

```diff
@@ -90,13 +91,31 @@ def _get_requested_booths(self, event, event_booth_ids):
     def event_booth_registration_confirm(self, event, booth_category_id, event_booth_ids, **kwargs):
         booths = self._get_requested_booths(event, event_booth_ids)
 
-        if not booths:
-            return json.dumps({'error': 'boothError'})
+        error_code = self._check_booth_registration_values(booths, kwargs['contact_email'])
+        if error_code:
+            return json.dumps({'error': error_code})
+
         booth_values = self._prepare_booth_registration_values(event, kwargs)
         booths.action_confirm(booth_values)
 
         return self._prepare_booth_registration_success_values(event.name, booth_values)
 
+    def _check_booth_registration_values(self, booths, contact_email, booth_category=False):
+        if not booths:
+            return 'boothError'
+
+        if booth_category and not booth_category.exists():
+            return 'boothCategoryError'
+
+        if request.env.user._is_public():
+            partner = request.env['res.partner'].sudo().search([
```

Comment:
Note that there are tools in mail for that purpose. I am in 18.2 and can't switch to 16, but look at stuff like "find partner from emails", stuff like that.